JASP-Desktop/data/labelmodel.cpp

```diff
@@ -117,3 +118,11 @@ void LabelModel::columnAboutToBeRemoved(int column)
 	if(proxyParentColumn() == column)
 		setVisible(false);
 }
+
+void LabelModel::columnDataTypeChanged(std::string colName)
+{
+	int colIndex = DataSetPackage::pkg()->getColumnIndex(colName);
+
+	if(colIndex == proxyParentColumn())
+		invalidate();
```

Comment:
I'm no sure I understand this invalidate(). It invalidates the current filter? But if I try it (I open the VaraiblesWindow, uncheck some labels, and change the type of the column), the filter still exists.

Reply:
the `invalidate()` is for the QSortFilterProxy, not our filter. It is to make sure that the new items are loaded if the columntype changes.
Normally it would reset after you change the `proxyParentColumn()` (which defines which parent column in the root of DataSetPackageis used). But if you have a column selected and the labelModel is set to it, you then change the type to "scale" and back to "ordinal" then `proxyParentColumn` doesn't change so nothing gets reloaded.

It would be better to emit some kind of `dataChanged` from DataSetPackage, but I couldn't get that to work so I did this.